subsys/bluetooth/services/ots/ots_client.c

```diff
@@ -498,6 +498,10 @@ int bt_ots_client_register(struct bt_ots_client *otc_inst)
 int bt_ots_client_unregister(uint8_t index)
 {
 	if (index < ARRAY_SIZE(otc_insts)) {
+		if (cur_inst == &otc_insts[index]) {
+			cur_inst = NULL;
```

Comment:
I would suggest to do
```suggestion
			return -EBUSY;
```

The only downside of my suggestion is that in case the the procedure is somehow halted, we cannot abort it.